app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java

```diff
@@ -0,0 +1,158 @@
+package com.ayush.weatherapp.home;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.support.design.widget.NavigationView;
+import android.support.v4.view.GravityCompat;
+import android.support.v4.widget.DrawerLayout;
+import android.support.v7.widget.Toolbar;
+import android.view.LayoutInflater;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+import android.widget.Toast;
+import butterknife.BindView;
+import com.ayush.weatherapp.R;
+import com.ayush.weatherapp.customViews.ForecastCompoundView;
+import com.ayush.weatherapp.customViews.ForecastDetailCompoundView;
+import com.ayush.weatherapp.retrofit.weatherApi.pojo.CurrentForecast;
+import com.ayush.weatherapp.retrofit.weatherApi.pojo.DailyForecast;
+import com.ayush.weatherapp.mvp.BaseActivity;
+import com.ayush.weatherapp.utils.DateUtils;
+import com.ayush.weatherapp.utils.LocationUtils;
+import com.ayush.weatherapp.mapper.WeatherImageMapper;
+import com.ayush.weatherapp.utils.MathUtils;
+import java.util.List;
+
+public class HomeActivity extends BaseActivity implements HomeContract.View {
+
+  @BindView(R.id.layout_drawer) DrawerLayout drawerLayout;
+  @BindView(R.id.nav_view) NavigationView navigationView;
+  @BindView(R.id.toolbar) Toolbar toolbar;
+
+  @BindView(R.id.tv_location) TextView tvLocation;
+  @BindView(R.id.tv_temperature_summary) TextView tvCurrentForecastSummary;
+  @BindView(R.id.tv_temp_current) TextView tvTempCurrent;
+  @BindView(R.id.iv_weather) ImageView ivWeather;
+
+  @BindView(R.id.grp_list_forecast) LinearLayout grpListForecast;
+
+  @BindView(R.id.detail_sun) ForecastDetailCompoundView detailSun;
+  @BindView(R.id.detail_wind) ForecastDetailCompoundView detailWind;
+  @BindView(R.id.detail_temperature) ForecastDetailCompoundView detailTemperature;
+
+  HomeContract.Presenter presenter;
+
+  @Override protected int getLayoutId() {
+    return R.layout.activity_home;
+  }
+
+  @Override protected void onCreate(Bundle savedInstanceState) {
+    //override theme defined in the xml for splash screen effect
+    setTheme(R.style.AppTheme);
+
+    super.onCreate(savedInstanceState);
+
+    initToolbar(toolbar);
+    showTitleBar(false);
+
+    setNavigationView();
+
+    presenter = new HomePresenterImpl(this);
```

Comment:
should be have a abstract method 
protected abstract void setupPresenter()
in base activity. Doing this ensures that we remember to setup presenter whenever we extend baseactivity

Reply:
:ok: